packages/aws-cdk-lib/aws-cloudfront-origins/lib/s3-origin.ts

```diff
@@ -44,46 +108,210 @@ export class S3Origin implements cloudfront.IOrigin {
+        }
+
+        const distributionId = options.distributionId;
+        const actions = this.getActions(props.originAccessLevels ?? [AccessLevel.READ]);
+        const result = this.grantDistributionAccessToBucket(distributionId, actions);
+
+        // Failed to update bucket policy, assume using imported bucket
+        if (!result.statementAdded) {
+          if (props.overrideImportedBucketPolicy) {
+            this.grantDistributionAccessToImportedBucket(scope, distributionId, actions);
+          } else {
+            Annotations.of(scope).addWarningV2('@aws-cdk/aws-cloudfront-origins:updateBucketPolicy',
+              'Cannot update bucket policy of an imported bucket. Set overrideImportedBucketPolicy to true or update the policy manually instead.');
+          }
+        }
+
+        if (this.bucket.encryptionKey) {
+          this.grantDistributionAccessToKey(
+            scope,
+            distributionId,
+            this.bucket.encryptionKey,
+            props.originAccessLevels ?? [AccessLevel.READ],
+          );
+        }
+
+        const originBindConfig = this._bind(scope, options);
+
+        // Update configuration to set OriginControlAccessId property
+        return {
+          ...originBindConfig,
+          originProperty: {
+            ...originBindConfig.originProperty!,
+            originAccessControlId: this.originAccessControl.originAccessControlId,
+          },
+        };
+      }
+
+      /**
+      * If you're using origin access control (OAC) instead of origin access identity, specify an empty `OriginAccessIdentity` element.
+      * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-s3originconfig.html#cfn-cloudfront-distribution-s3originconfig-originaccessidentity
+      */
+      protected renderS3OriginConfig(): cloudfront.CfnDistribution.S3OriginConfigProperty | undefined {
+        return { originAccessIdentity: '' };
+      }
+
+      private getActions(accessLevels: AccessLevel[]): string[] {
+        let actions: string[] = [];
+        for (const accessLevel of new Set(accessLevels)) {
+          actions = actions.concat(BUCKET_ACTIONS[accessLevel]);
+        }
+        return actions;
+      }
+
+      private grantDistributionAccessToBucket(distributionId: string | undefined, actions: string[]): iam.AddToResourcePolicyResult {
+        const oacReadOnlyBucketPolicyStatement = new iam.PolicyStatement(
+          {
+            sid: 'GrantOACAccessToS3',
+            effect: iam.Effect.ALLOW,
+            principals: [new iam.ServicePrincipal('cloudfront.amazonaws.com')],
+            actions,
+            resources: [this.bucket.arnForObjects('*')],
+            conditions: {
+              StringEquals: {
+                'AWS:SourceArn': `arn:${Aws.PARTITION}:cloudfront::${Aws.ACCOUNT_ID}:distribution/${distributionId}`,
```

Comment:
For my learning purposes, is CloudFront arn not including region?

Reply:
Correct, CloudFront Distribution arns don't have the region since they are considered global resources